src/librustc/ty/context.rs

```diff
@@ -1942,8 +1942,12 @@ pub mod tls {
     /// This is a callback from libsyntax as it cannot access the implicit state
     /// in librustc otherwise
     fn span_debug(span: syntax_pos::Span, f: &mut fmt::Formatter<'_>) -> fmt::Result {
```

Comment:
I am confused, isn't this only used *at the same time* as the TLS `TyCtxt` is present?
How can `with_opt` give you `None`? Did that code change somehow?

Reply:
With parallel queries we set this on the worker threads when we create the thread pool. In that case it acts it's more like runtime linking and it will be set when no `TyCtxt` is around.

Reply:
Why not set it at the same time as the `TyCtxt` TLS?

Reply:
We need to set this on all worker threads, which we can easily do when creating the thread pool.
We could put it in `ImplicitCtxt`, but it would never change, so that would just make queries slower for no reason.

Reply:
In that case, can we make it so the debug callbacks are *all* set globally before entering even the global `TyCtxt` even without parallel queries?

Reply:
Maybe we just need a `scoped_thread_local` for the `CodeMap`? I feel like that would probably be a cleaner design.